src/mon/MgrStatMonitor.cc

```diff
@@ -101,6 +101,15 @@ void MgrStatMonitor::create_initial()
   encode(service_map, pending_service_map_bl, CEPH_FEATURES_ALL);
 }
 
+void MgrStatMonitor::clear_pool_availability(int64_t poolid)
+{
+  dout(20) << __func__ << dendl;
+  std::scoped_lock l(lock);
+  pending_pool_availability.erase(poolid);
+  pending_pool_availability.insert({poolid, PoolAvailability()});
```

Comment:
I think pending_pool_availability here is a map, In my humble opinion, erasing the key-value and then inserting it back again is not so elegant, since probably it will re-hash the new key, and also what if the pool has been erased (I know that you already put a lock here, but I think the following might be a better engineering practice):

auto pool_itr = pending_pool_availabiltiy.find(poolid);
if (pool_itr != pending_pool_availability.end()) {
   pool_itr->second = PoolAvailability();
} else {
   dout(1) << "failed to clear a non-existing pool: " << poolid << dendl;
}


performance wise probably still O(logn) (since red-black tree) for both your way and mine, it's just this way might mitigate certain unexpected behavior like pool getting deleted.